### PR TITLE
test: unittest_hostname compile error on freebsd

### PR DESCRIPTION
**Pull request by Liuchang0812**
Signed-off-by: liuchang0812 <liuchang0812@gmail.com>

@wjwithagen I have no FreeBSD environment, mind taking a look? It's compiled successfully by clang. thanks!

### CODE REVIEW

File: src/test/common/test_hostname.cc [hunk 1]
```diff
@@ -22,8 +22,9 @@
 
 #include "unistd.h"
 
-#include <string>
+#include <array>
 #include <iostream>
+#include <string>
```
Comment (tchaikov) [inline src/test/common/test_hostname.cc]: might want to remove `#include <string>`.

Comment (Liuchang0812) [inline src/test/common/test_hostname.cc]: but std::string is used in line 35

Comment (Liuchang0812) [inline src/test/common/test_hostname.cc]: oh, do you mean that I shouldn't reorder `#include <string>` ?

**Comment by Liuchang0812** [comment]
```
../../lib/libceph-common.so.0: undefined reference to `boost::re_detail_106300::cpp_regex_traits_implementation<char>::transform[abi:cxx11](char const*, char const*) const'
../../lib/libceph-common.so.0: undefined reference to `boost::re_detail_106300::cpp_regex_traits_implementation<char>::transform_primary[abi:cxx11](char const*, char const*) const'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
src/test/CMakeFiles/unittest_xlist.dir/build.make:117: recipe for target 'bin/unittest_xlist' failed
make[2]: *** [bin/unittest_xlist] Error 1
CMakeFiles/Makefile2:7156: recipe for target 'src/test/CMakeFiles/unittest_xlist.dir/all' failed
make[1]: *** [src/test/CMakeFiles/unittest_xlist.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Seems that clang linker could not find boost::regex.

File: src/test/common/test_hostname.cc (hunk 1 above)
Comment (tchaikov) [inline src/test/common/test_hostname.cc]: it's duplicated with the `#include` at line:30

Comment (Liuchang0812) [inline src/test/common/test_hostname.cc]: sorry. fixed it

**Comment by Liuchang0812** [comment]
@wjwithagen  thanks!

**Pull request merged**